### PR TITLE
Code Style - apply PascalCase

### PR DESCRIPTION
**Pull request by chriselion**
### Proposed change(s)

Unity's C# style guide says
```
In the Unity namespace, public fields are PascalCase with no prefix
In the Unity namespace, public properties are PascalCase with no prefix
```
(as opposed to UnityEngine and UnityEditor namespaces, which have different conventions).

This PR updates all the public fields and properties that I could find (via API scraper) that were camelCase instead of PascalCase

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)
https://ono.unity3d.com/unity-extra/unity-meta/files/@/ReferenceSource/CSharp/Assets/CSharpReference.cs

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: com.unity.ml-agents/Runtime/Sensors/CameraSensorComponent.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public bool grayscale
         /// <summary>
         /// The compression type to use for the sensor.
         /// </summary>
-        public SensorCompressionType compression
+        public SensorCompressionType CompressionType
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensors/CameraSensorComponent.cs:84]: I renamed a few of these to be more consistent too.

**Comment by vincentpierre** [comment]
There should be some changes to the documentation I think.
https://github.com/Unity-Technologies/ml-agents/blame/api-PascalCase/docs/Learning-Environment-Examples.md#L364

**Comment by chriselion** [comment]
😭  Yeah, I'll do a pass on those too.

**Comment by chriselion** [comment]
I skimmed through code blocks, and did a few spot checks with `git grep foo -- docs/.` and didn't find any others:
```
$ git grep maxStep -- docs/.
docs/Migrating.md:  instead of "camelCase"; for example, `Agent.maxStep` was renamed to
docs/Migrating.md:    `maxStep` information)
docs/Migrating.md:  - `maxStep` is now a public field on the Agent. (Was moved from
```

**Comment by mmattar** [comment]
@chriselion - is there any error-ing/warning/pre-commit checks we can do to defend against this in the future?

**Comment by chriselion** [comment]
@mmattar Nothing that I'm aware of. It sounds like something the [Coding Support Tools package] (https://docs.unity3d.com/Packages/com.unity.coding@0.1/manual/index.html) could potentially handle but not sure if it does (and we don't currently have a way to enforce anything else it does like code formatting anyway). 

More extreme, we could use that package to generate a dump of the API scraping and parse that.

**Comment by chriselion** [comment]
Followup for enforcing this automatically: https://jira.unity3d.com/browse/MLA-947

**Comment by JPhilipp** [comment]
> Unity's C# style guide

@chriselion Is this document publicly available by any chance? I'd be interested to have a look (I understand it's also Microsoft C# convention to go for .PascalCase for public fields, whereas I'm always used to Unity fields being .camelCase).

**Comment by chriselion** [comment]
@JPhilipp I'll find out - AFAIK the only version is in Unity's internal repo, but there might be a public copy somewhere.

**Comment by chriselion** [comment]
@JPhilipp You can install the `com.unity.coding` package (it's in the public registry, but not visible by default), and once you do that, there's a copy of the file in 
`{your project}/Library/PackageCache/com.unity.coding@0.1.0-preview.13/Coding~/Conventions/CSharpReference.cs`

Not exactly obvious :/ but hopefully that's enough to go on. Here's the line to add it to your list of dependencies: 
https://github.com/Unity-Technologies/ml-agents/blob/eedc3f9c052295d89bed0ac40a8e82a8fd17fead/DevProject/Packages/manifest.json#L7

**Pull request merged**